src/fastgltf.cpp

```diff
@@ -2203,6 +2251,7 @@ bool fg::GltfDataBuffer::loadFromFile(const fs::path& path, std::uint64_t byteOf
     file.read(reinterpret_cast<char*>(bufferPointer), static_cast<std::streamsize>(dataSize));
     std::memset(bufferPointer + dataSize, 0, allocatedSize - dataSize);
     return true;
+    #endif
```

Comment:
I strongly thing this is wrong. What if you are on Android but want to use std::ifstream for whatever reason? Also, one could call the default constructor and then the asset manager would be nullptr, causing a segfault here.

Instead, I think the `loadFromFile_Android` should be called `loadFromFileAndroid` and be public, so that its use is explicit. (And that should have an `assert(asset_manager != nullptr)`

Reply:
I've moved the Android loading functionality to a child class. The only way to construct it is with an `AAssetManager`. This will help prevent people from loading an Android asset without an `AAssetManager`

I've renamed the new method to `loadFromAndroidAsset` since technically the assets in the APK aren't files. You can still use `loadFromFile` to load a glTF from a file external to your APK

Reply:
Looks good. Then I'd guess `AndroidGltfDataBuffer` should make `loadFromFile` private, because it wouldn't make sense to have those functions available to a struct made exclusively to load APK assets? So, effectively this:
```cpp
class AndroidGltfDataBuffer : public GltfDataBuffer {
private:
    using GltfDataBuffer::loadFromFile;
}
```

Reply:
I don't think it should be private. On Android, you can still load a file from outside the APK using standard C++ classes

Reply:
In that case you can just use `GltfDataBuffer`, no?

Reply:
I don't think so. I wanted to make a subclass so that you could only try to load an Android asset if you've supplied an `AAssetManager`